### PR TITLE
Drawer Improvement, Let define a route key to open/close multiple dra…

### PR DESCRIPTION
**Pull request by flo-pereira**
This PR simply add a drawerOpen/drawerClose navigation keys that let add multiple loaders without side effects

Solve this kind of issue #780

All drawer with same key will open close simultanly

See examples/NavigationPlayground/js/MultipleDrawer.js to see it in action
```

const DrawerExample = DrawerNavigator(
  {
    Inbox: {
      path: '/',
      screen: InboxScreen,
    },
    Drafts: {
      path: '/sent',
      screen: DraftsScreen,
    },
  },
  {
    initialRouteName: 'Drafts',
    contentOptions: {
      activeTintColor: '#e91e63',
    },
  }
);

const MainDrawerExample = DrawerNavigator({
  Drafts: {
    screen: DrawerExample,
  },
}, {
  drawerOpenRoute: 'FooDrawerOpen', //your custom navigation key (default: DrawerOpen)
  drawerCloseRoute: 'FooDrawerClose',
});
```

```
//For the default drawer 
navigation.navigate('DrawerOpen')

//For your custom drawer(s)
navigation.navigate('FooDrawerOpen')
```

### CODE REVIEW

**Comment by borisyankov** [comment]
This is a great contribution. Would love to use the multi-drawers in our app. 👍

**Comment by XeniousSoftware** [comment]
i think this needs to be merged ASAP. very useful solution.

**Comment by PavelPZ** [comment]
Thanks a lot :-)
Waiting for it.

**Comment by johntran** [comment]
This would be an awesome improvement.

**Comment by mendyismyname** [comment]
Was so excited to use this feature. Hopefully it gets merged soon!

**Comment by chmac** [comment]
@flo-pereira Thanks for this, it's something that will be super useful for us once merged.

Any update on the status of this PR? I see a couple of red "Review required" and "All checks have failed" above. Looks like there's a merge conflict which probably needs resolved before this could be merged.

Any feedback from the maintainers on if / when this will be merged? We'd like to use this production in 2-3 months so trying to get a feel for a realistic timeline.

**Comment by flo-pereira** [comment]
I'll try to find time this week to deal with this.

**Comment by flo-pereira** [comment]
tests passed!

**Comment by kelset** [comment]
Thanks for submitting, pinging maybe @GantMan or @vonovak which could be good "fresh" reviewers for it.

Sadly can't give any specific guidelines for when it will be merged.

**Comment by hoscarcito** [comment]
+1

**Comment by flo-pereira** [comment]
thanks

**Comment by LeoLaiFookSiang** [comment]
Hi, may I know can I create 2 drawer object in a same variable ? For example:

const DrawerExample = DrawerNavigator(
  {
    Home: { screen: Home },
    Anatomy: { screen: Anatomy },
  },
  {
    initialRouteName: "Home",
    contentOptions: {
      activeTintColor: "#e91e63"
    },
    contentComponent: props => <SideBar {...props} />
  },
  {
    drawerPosition: "right",
    contentOptions: {
      activeTintColor: "#e91e63"
    },
    drawerOpenRoute: 'FooDrawerOpen',
    drawerCloseRoute: 'FooDrawerClose',
    contentComponent: props => <SideBar {...props} />
  },
);

**Comment by Longsight** [comment]
Note that as a result of this change, if you define a `DrawerNavigator` with an options object, that object must now have, at the very least, a `DrawerCloseRoute` property - and probably `Open` and `Toggle` as well, if you ever intend to use them. DrawerNavigators defined without these properties will fail to correctly select the initial DrawerClose route, and bail out.

File: src/navigators/DrawerNavigator.js [hunk 1]
```diff
@@ -82,14 +86,17 @@ const DrawerNavigator = (
     drawerPosition,
     useNativeAnimations,
     drawerBackgroundColor,
+    drawerOpenRoute,
```
Comment (smaclell) [inline src/navigators/DrawerNavigator.js]: With this change these three properties are mandatory if you override the `config`. This is due to `config` overriding the `DefaultDrawerConfig`. Was that intentional?

**Comment by fruedaCode** [comment]
Is this option still available in version 3??

**Pull request merged**